### PR TITLE
Add initial community analyzers

### PR DESCRIPTION
**Pull request by srijan-deepsource**
Ref: ENG-36

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+steps:
+  - name: 'curlimages/curl'
+    args:
+      [
+        '--location',
+        '--request', 'POST',
+        '--header', "Authorization: Bearer ${_DATASYNC_SECRET}",
+        '--header', 'Content-Type: application/json',
+        '--data-raw', '{"query":" mutation($input: TriggerAnalyzerSyncInput!){\n    triggerAnalyzerSync(input: $input){\n        ok\n    }\n}","variables":{"input":{"repoName":"community-analyzers","tag":"$TAG_NAME", "analyzer_version_map": {"infer": "v1.1.0", "kube-linter": "0.6.4"}}}}',
+        'https://deepsource.io/internal/graphql/'
```
Comment (srijan-deepsource) [inline cloudbuild.yaml]:
```suggestion
        'https://app.deepsource.one/internal/graphql/'
```

File: cloudbuild.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+steps:
+  - name: 'curlimages/curl'
+    args:
+      [
+        '--location',
+        '--request', 'POST',
+        '--header', "Authorization: Bearer ${_DATASYNC_SECRET}",
+        '--header', 'Content-Type: application/json',
+        '--data-raw', '{"query":" mutation($input: TriggerAnalyzerSyncInput!){\n    triggerAnalyzerSync(input: $input){\n        ok\n    }\n}","variables":{"input":{"repoName":"community-analyzers","tag":"$TAG_NAME", "analyzer_version_map": {"infer": "v1.1.0", "kube-linter": "0.6.4"}}}}',
```
Comment (srijan-deepsource) [inline cloudbuild.yaml]:
```suggestion
        '--data-raw', '{"query":" mutation($input: TriggerAnalyzerSyncInput!){\n    triggerAnalyzerSync(input: $input){\n        ok\n    }\n}","variables":{"input":{"repoName":"community-analyzers","tag":"$TAG_NAME","analyzer_version_map":{\"infer\":\"v1.1.0\",\"kube-linter\":\"0.6.4\"}}}}',
```

File: cloudbuild_dev.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+steps:
+  - name: 'curlimages/curl'
+    args:
+      [
+        '--location',
+        '--request', 'POST',
+        '--header', "Authorization: Bearer ${_DATASYNC_SECRET}",
+        '--header', 'Content-Type: application/json',
+        '--data-raw', '{"query":" mutation($input: TriggerAnalyzerSyncInput!){\n    triggerAnalyzerSync(input: $input){\n        ok\n    }\n}","variables":{"input":{"repoName":"community-analyzers","tag":"$TAG_NAME", "analyzer_version_map": {"infer": "v1.1.0", "kube-linter": "0.6.4"}}}}',
```
Comment (srijan-deepsource) [inline cloudbuild_dev.yaml]:
```suggestion
        '--data-raw', '{"query":" mutation($input: TriggerAnalyzerSyncInput!){\n    triggerAnalyzerSync(input: $input){\n        ok\n    }\n}","variables":{"input":{"repoName":"community-analyzers","tag":"$TAG_NAME","analyzer_version_map": {\"infer\":\"v1.1.0\",\"kube-linter\":\"0.6.4\"}}}}',
```

File: analyzers/kube-linter/.deepsource/analyzer/analyzer.toml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Analyzer Profile
+category = "conf"
+name = "KubeLinter"
+shortcode = "kube-linter"
+status = "active"
+processors = ["skipcq", "source_code_load"]
```
Comment (srijan-deepsource) [inline analyzers/kube-linter/.deepsource/analyzer/analyzer.toml:6]:
```suggestion
processors = ["skip_cq", "source_code_load"]
```

Comment (srijan-deepsource) [inline analyzers/kube-linter/.deepsource/analyzer/analyzer.toml]:
```suggestion
```

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
+*$py.class
```
Comment (tushar-deepsource) [inline .gitignore:4]:
```suggestion
*$py.class
venv*
.tox
.coverage
```

File: analyzers/kube-linter/utils/issue_map.json [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    "use-namespace": {
+        "issue_code": "KUBELIN-W1049"
+    },
+    "wildcard-in-rules": {
+        "issue_code": "KUBELIN-W1050"
+    },
+    "writable-host-mount": {
+        "issue_code": "KUBELIN-W1051"
+    }
+}
```
Comment (tushar-deepsource) [inline analyzers/kube-linter/utils/issue_map.json]:
```suggestion
}

```

File: analyzers/infer/utils/issue_scraper.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    with TemporaryDirectory() as temp_dir_name:
+        subprocess.run(clone_command, cwd=temp_dir_name)
+        resp = subprocess.run(["ls", temp_dir_name])
+        infer_issues_dir = f"{temp_dir_name}/infer/infer/documentation/issues"
+        for filename in os.listdir(infer_issues_dir):
+            if not filename.endswith(".md"):
+                pass
+            infer_issue_id = filename.removesuffix(".md")
+
+        breakpoint()
```
Comment (tushar-deepsource) [inline analyzers/infer/utils/issue_scraper.py]: ?

Comment (srijan-deepsource) [reply analyzers/infer/utils/issue_scraper.py]: oops

File: analyzers/infer/utils/issue_scraper.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+ISSUE_MAP = ""
+CLONE_URL = "https://github.com/facebook/infer.git"
+
+
+def clone_issues_locally():
+    """Clone the issues directory from infer's GitHub repo."""
+    clone_command = f"git clone {CLONE_URL} --depth=1".split()
+    with TemporaryDirectory() as temp_dir_name:
+        subprocess.run(clone_command, cwd=temp_dir_name)
+        resp = subprocess.run(["ls", temp_dir_name])
```
Comment (tushar-deepsource) [inline analyzers/infer/utils/issue_scraper.py]:
```suggestion
```

File: analyzers/kube-linter/utils/issue_map_gen.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    # run kubelinter to get the list of checks in the json format
+    cmd = "kube-linter checks list --format json".split()
+    resp = subprocess.run(cmd, capture_output=True)
+    return json.loads(resp.stdout.decode("utf-8"))
+
+
+def get_next_code(mapping) -> str:
+    """Return the next available issue code."""
+    num_issues = len(mapping.keys())  # get the number of issues already in the mapping
+    next_code = 1000 + num_issues + 1  # issue code series starts from `1001`
```
Comment (tushar-deepsource) [inline analyzers/kube-linter/utils/issue_map_gen.py]:
```suggestion
    next_code = 1001 + num_issues  # issue code series starts from `1001`
```

File: analyzers/kube-linter/utils/issue_map_gen.py [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    return json.loads(resp.stdout.decode("utf-8"))
+
+
+def get_next_code(mapping) -> str:
+    """Return the next available issue code."""
+    num_issues = len(mapping.keys())  # get the number of issues already in the mapping
+    next_code = 1001 + num_issues  # issue code series starts from `1001`
+    while True:
+        yield next_code
+        next_code += 1
```
Comment (tushar-deepsource) [inline analyzers/kube-linter/utils/issue_map_gen.py]:
```suggestion
    yield from itertools.count(next_code)
```

File: analyzers/kube-linter/utils/issue_map_gen.py [hunk 11]
```diff
@@ -0,0 +1,53 @@
+import subprocess
+import os
+import json
```
Comment (tushar-deepsource) [inline analyzers/kube-linter/utils/issue_map_gen.py]:
```suggestion
import itertools
import json
import os
import subprocess
```

File: analyzers/infer/utils/issue_scraper.py [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+def clone_issues_locally():
+    """Clone the issues directory from infer's GitHub repo."""
+    clone_command = f"git clone {CLONE_URL} --depth=1".split()
+    with TemporaryDirectory() as temp_dir_name:
+        subprocess.run(clone_command, cwd=temp_dir_name)
+        infer_issues_dir = f"{temp_dir_name}/infer/infer/documentation/issues"
+        for filename in os.listdir(infer_issues_dir):
+            if not filename.endswith(".md"):
+                pass
```
Comment (tushar-deepsource) [inline analyzers/infer/utils/issue_scraper.py]:
```suggestion
                continue
```

**Review by tushar-deepsource — APPROVED** [review]
LGTM

**Pull request merged**